browser/ui/views/infobars/brave_wayback_machine_infobar_contents_view.cc

```diff
@@ -176,17 +186,55 @@ void BraveWaybackMachineInfoBarContentsView::InitializeChildren() {
                   0));
   AddChildView(label);
 
-  button_ = new BraveWaybackMachineInfoBarButtonContainer(base::BindRepeating(
-      &BraveWaybackMachineInfoBarContentsView::ButtonPressed,
-      base::Unretained(this)));
-  views_visible_before_checking_.push_back(button_);
-  button_->SetProperty(
-      views::kMarginsKey,
-      gfx::Insets(ChromeLayoutProvider::Get()->GetDistanceMetric(
-                      DISTANCE_TOAST_CONTROL_VERTICAL),
-                  0));
-  button_->SizeToPreferredSize();
-  AddChildView(button_);
+  auto dont_ask_button = std::make_unique<views::MdTextButton>(
+      base::BindRepeating(
+          &BraveWaybackMachineInfoBarContentsView::DontAskButtonPressed,
+          base::Unretained(this)),
+      l10n_util::GetStringUTF16(IDS_BRAVE_WAYBACK_MACHINE_DONT_ASK_AGAIN_TEXT));
+  dont_ask_button_ = dont_ask_button.get();
+  views_visible_before_checking_.push_back(dont_ask_button_);
+
+  fetch_url_button_ =
+      new BraveWaybackMachineInfoBarButtonContainer(base::BindRepeating(
+          &BraveWaybackMachineInfoBarContentsView::FetchURLButtonPressed,
+          base::Unretained(this)));
+  views_visible_before_checking_.push_back(fetch_url_button_);
+
+  if (views::PlatformStyle::kIsOkButtonLeading) {
+    fetch_url_button_->SetProperty(
+        views::kMarginsKey,
+        gfx::Insets(ChromeLayoutProvider::Get()->GetDistanceMetric(
+                        DISTANCE_TOAST_CONTROL_VERTICAL),
+                    ChromeLayoutProvider::Get()->GetDistanceMetric(
+                        DISTANCE_RELATED_CONTROL_HORIZONTAL_SMALL)));
+    fetch_url_button_->SizeToPreferredSize();
+    AddChildView(fetch_url_button_);
+
+    dont_ask_button_->SetProperty(
+        views::kMarginsKey,
+        gfx::Insets(ChromeLayoutProvider::Get()->GetDistanceMetric(
+                        DISTANCE_TOAST_CONTROL_VERTICAL),
+                    0));
+    dont_ask_button_->SizeToPreferredSize();
+    AddChildView(dont_ask_button.release());
```

Comment:
`dont_ask_button_ = AddChildView(std::move(dont_ask_button));` would be better to show ownership transfer.

Reply:
@rahulsnkr I tried some cleanup - https://github.com/brave/brave-core/pull/9069/commits/70dfe0a79ae38f0aa49889e7557bcf663f839dc9 WDYT?

Reply:
@simonhong LGTM, thanks :)

Reply:
@rahulsnkr Ok, then please cherry-pick https://github.com/brave/brave-core/commit/70dfe0a79ae38f0aa49889e7557bcf663f839dc9 to your branch.
I triggered CI for this PR with https://github.com/brave/brave-core/pull/9069 and will merge this PR when it has all green :)